src/main/java/seedu/address/logic/filters/NameFilter.java

```diff
@@ -29,50 +29,37 @@ public NameFilter(String nameListSingleString) {
         checkArgument(isValidFilter(nameListSingleString), MESSAGE_CONSTRAINTS);
         this.nameList = nameListSingleString.split("\\s+");
     }
+
     /**
      * Returns true if a given string is a valid filter.
      */
     public static boolean isValidFilter(String filterString) {
         return filterString.matches(VALIDATION_REGEX);
     }
 
-    @Override
-    public boolean test(Customer customer) {
-        requireNonNull(customer);
-        String[] customerNameTokens = customer.getName().fullName.split("\\s+");
-        for (String token : customerNameTokens) {
-            for (String possibleName : nameList) {
-                if (levenshteinDistance(token, possibleName) <= MISTAKE_THRESHOLD) {
-                    return true;
-                }
-            }
-        }
-        return false;
-    }
-
     //@@author nighoggDatatype-reused
     //Reused from https://stackoverflow.com/a/13564498/11358676
-    private static int levenshteinDistance(String s1, String s2) {
-        return dist(s1.toCharArray(), s2.toCharArray());
+    private boolean levenshteinDistance(String s1, String s2) {
+        return isSubsequence(s1.toCharArray(), s2.toCharArray());
```

Comment:
I suppose that this is better than my current implementation, but after you commit, I will provide a potentially better version.

Reply:
Or at least try to